scoap3/articles/tasks.py

```diff
@@ -122,6 +124,20 @@ def check_doi_registration_time(obj):
     return False, "DOI not found in our system."
 
 
+def check_authors_affiliation(article_id):
+    try:
+        article = Article.objects.get(id=article_id)
+        authors = Author.objects.filter(article_id=article)
+        for author in authors:
+            affiliations = Affiliation.objects.filter(author_id=author)
+            if len(affiliations) < 1:
+                return False, "Author does not have affiliations"
+        return True, "Authors' affiliations are compliant"
+    except Article.DoesNotExist:
```

Comment:
we shoudnt need `Article.DoesNotExist`, else there would be no compliance check to be initiated (should be caught from before)